compat/wserrno.cc

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/wserrno.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+
+#include <unordered_map>
+
+void
+SetErrnoFromWsaError()
+{
+    // POSIX codes which socket API users may care about
+    static const auto *CodeMap = new std::unordered_map<int, int> {
+
+        { WSAECONNABORTED, ECONNABORTED },
+
+        { WSAEINPROGRESS, EINPROGRESS },
+        { WSAEAFNOSUPPORT, EAFNOSUPPORT },
+        { WSAEINVAL, EINVAL },
+        { WSAEISCONN, EISCONN },
+
+        { WSAEWOULDBLOCK, EWOULDBLOCK },
+        // Windows does not have a equivalent of EAGAIN
```

Comment:
To match the other two similar comments (and fix grammar):

```suggestion
        // no Windows error code maps to EAGAIN
```

Reply:
ok